Cython/Utility/ObjectHandling.c

```diff
@@ -72,10 +72,8 @@ static int __Pyx_IternextUnpackEndCheck(PyObject *retval, Py_ssize_t expected) {
         Py_DECREF(retval);
         __Pyx_RaiseTooManyValuesError(expected);
         return -1;
-    } else {
-        return __Pyx_IterFinish();
     }
-    return 0;
+    return __Pyx_IterFinish();
```

Comment:
```suggestion

    return __Pyx_IterFinish();
```

Extra line to match current master

Reply:
Done.